api-lib/event_triggers/removeTeammate.ts

```diff
@@ -0,0 +1,124 @@
+import assert from 'assert';
+
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+
+import { ValueTypes } from '../gql/__generated__/zeus';
+import { adminClient } from '../gql/adminClient';
+import { errorResponse } from '../HttpError';
+import { EventTriggerPayload } from '../types';
+
+export default async function handler(req: VercelRequest, res: VercelResponse) {
+  // no parsing should be needed here since this data comes straight from
+  // the database and zeus keeps this consistent for us
+  const {
+    event: { data },
+  }: EventTriggerPayload<'teammates', 'DELETE'> = req.body;
+
+  const { user_id, team_mate_id } = data.old;
+
+  const results = [];
+  try {
+    const { users_by_pk } = await adminClient.query({
+      users_by_pk: [
+        {
+          id: user_id,
+        },
+        {
+          pending_sent_gifts: [
+            {
+              where: {
+                recipient_id: {
+                  _eq: team_mate_id,
+                },
+              },
+            },
+            {
+              id: true,
+              recipient_id: true,
+              tokens: true,
+            },
+          ],
+          circle: {
+            epochs: [
+              {
+                where: {
+                  end_date: { _gt: 'now()' },
+                  start_date: { _lt: 'now()' },
+                },
+              },
+              {
+                id: true,
+              },
+            ],
+          },
+        },
+      ],
+    });
+    assert(users_by_pk, 'panic: user must exist');
+
+    const { pending_sent_gifts, circle } = users_by_pk;
+
+    const currentEpoch = circle.epochs.pop();
+    if (!currentEpoch || pending_sent_gifts.length === 0) {
+      res.status(200).json({
+        message: `Not a refund event.`,
+      });
+      return;
+    }
+
+    const totalRefund = pending_sent_gifts
+      .map(gift => gift.tokens)
+      .reduce((total, tokens) => tokens + total);
+
+    const refundFromCounterpartyMutations = pending_sent_gifts.reduce(
+      (ops, gift) => {
+        if (gift.tokens > 0)
+          ops[gift.id] = {
+            update_users_by_pk: [
+              {
+                pk_columns: { id: gift.recipient_id },
+                _inc: { give_token_received: -gift.tokens },
+              },
+              { give_token_received: true },
+            ],
+          };
+        return ops;
+      },
+      {} as { [aliasKey: number]: ValueTypes['mutation_root'] }
+    );
```

Comment:
looks like we might want to create a library function suite for this pattern, haha